src/accessibility/context_menu/ContextMenuButton.tsx

```diff
@@ -16,25 +16,20 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from "react";
+import React, { forwardRef, Ref } from "react";
 
-import AccessibleButton from "../../components/views/elements/AccessibleButton";
+import AccessibleButton, { Props as AccessibleButtonProps } from "../../components/views/elements/AccessibleButton";
 
-interface IProps extends React.ComponentProps<typeof AccessibleButton> {
+type Props<T extends keyof JSX.IntrinsicElements> = AccessibleButtonProps<T> & {
     label?: string;
     // whether or not the context menu is currently open
     isExpanded: boolean;
-}
+};
 
-// Semantic component for representing the AccessibleButton which launches a <ContextMenu />
```

Comment:
As with `AccessibleButton`: since we're changing the name of the ref property, can we take the opportunity to document it? I shouldn't have to dig through a hierarchy of complex code to understand that (a) it exposes a ref property or (b) what said ref ends up pointed to.

Likewise on `ContextMenuTooltipButton` and `AccessibleTooltipButton`.

Reply:
Refs are a fundamental of React - https://react.dev/learn/referencing-values-with-refs The types state that it references a HTMLElement

<img width="563" alt="image" src="https://github.com/matrix-org/matrix-react-sdk/assets/2403652/2415704c-2e67-4d46-9c87-b6392d14762c">